gateway/crates/federated-server/src/server/gateway.rs

```diff
@@ -170,29 +178,11 @@ pub(super) async fn generate(
                 tls,
             };
 
-            match config_path {
-                Some(path) if config_hot_reload => {
-                    hot_reload::ConfigWatcher::new(path, tx).watch()?;
-                }
-                _ => (),
-            }
-
-            RedisRateLimiter::runtime(global_config, rx)
+            RedisRateLimiter::runtime(global_config, rate_limit_rx)
```

Comment:
:zap: **Suggestion** 
Ensure that the `rate_limit_rx` variable is properly handled in case of errors during runtime.